modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/ToolsEndpoint.java

```diff
@@ -370,14 +406,131 @@ public Response getVideoEditor(@PathParam("mediapackageid") final String mediaPa
               f("displayOrder", v(workflow.getDisplayOrder()))));
     }
 
+    // Get thumbnail
+    final List<Field> thumbnailFields = new ArrayList<>();
+    try {
+      final ThumbnailImpl thumbnailImpl = newThumbnailImpl();
+      final Optional<ThumbnailImpl.Thumbnail> optThumbnail = thumbnailImpl
+        .getThumbnail(mp, urlSigningService, expireSeconds);
+
+      optThumbnail.ifPresent(thumbnail -> {
+        thumbnailFields.add(f("type", thumbnail.getType().name()));
+        thumbnailFields.add(f("url", thumbnail.getUrl().toString()));
+        thumbnailFields.add(f("defaultPosition",  thumbnailImpl.getDefaultPosition()));
+        thumbnail.getPosition().ifPresent(p -> thumbnailFields.add(f("position", p)));
+        thumbnail.getTrack().ifPresent(t -> thumbnailFields.add(f("track", t)));
+      });
+    } catch (UrlSigningException | URISyntaxException e) {
+      logger.error("Error while trying to serialize the thumbnail url because: {}", getStackTrace(e));
+      throw new WebApplicationException(e, SC_INTERNAL_SERVER_ERROR);
+    }
+
+    final MediaPackageElementFlavor sourceTracksFlavor = new MediaPackageElementFlavor("*",
+      adminUIConfiguration.getThumbnailSourceFlavorSubtype());
+
+    final List<JValue> sourceTracks = Arrays.stream(mp.getTracks(sourceTracksFlavor))
+          .map(MediaPackageElement::getFlavor)
+          .map(e -> {
+            String side = null;
+            if (e.equals(this.adminUIConfiguration.getSourceTrackLeftFlavor())) {
```

Comment:
I am not sure if there are any configurations in which you may have more than 2 tracks. Just to be extra cautious, I have now set `side` to `"unknown"` in case the track doesn't have the left, nor the right source track flavor.